### PR TITLE
ETL-3251 Fix incorrect handling of column names containing 'where'

### PR DESCRIPTION
**Pull request by chavdar**
Changes: 

* Add a new helper class SqlQueryHelper and move JdbcExtractor.addPredicate to SqlQueryHelper
* Add some sanity checks for SqlQueryHelper.addPredicate()
* Add unit tests for SqlQueryHelper.addPredicate()
* Get rid of Utils.getClause() since it's very specific; I've inlined the implementation in SqlQueryHelper.addPredicate()
* Refactor JdbcExtractor and SalesforceExtractor to use SqlQueryHelper.addPredicate()

### CODE REVIEW

**Comment by chavdar** [comment]
@zliu41 can you please review?

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/**
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java]: Missing license header

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * A helper class for handling SQL queries as strings. This is a temp solution until something more
+ * robust is implemented.
+ *
+ * @see https://github.com/linkedin/gobblin/issues/236
+ */
+public class SqlQueryHelper {
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java]: Seems most of this types of classes in Gobblin are named "xxxUtils"

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+   *
+   * <b>Note that this method is rather limited. It works only if there are no other clauses after
+   * "where"</b>
+   *
+   * @param query           the query string to modify
+   * @param predicateCond   the predicate to add to the query
+   * @return query          the new query string
+   * @throws IllegalArgumentException if the predicate cannot be added because of additional clauses
+   */
+  public static String addPredicate(String query, String predicateCond) {
```
Comment (zliu41) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java]: The original code checks if `predicateCond` is null or empty. It is no longer checked here.

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java (hunk 1 above)
Comment (chavdar) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java]: Fixed.

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java (hunk 2 above)
Comment (chavdar) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java]: Renamed.

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java (hunk 3 above)
Comment (chavdar) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/SqlQueryHelper.java]: Good point. Fixed.

**Comment by zliu41** [comment]
LGTM

**Pull request merged**